azkaban-common/src/main/java/azkaban/server/HttpRequestUtils.java

```diff
@@ -366,21 +364,25 @@ public static Map<String, String> getParamGroup(final HttpServletRequest request
    * Read params like groupName[level1Key][level2Key]: value
    */
   public static Map<String, Map<String, String>> getMapParamGroup(final HttpServletRequest request,
-      final String groupName) {
+      final String groupName) throws ServletException {
     final Enumeration<String> enumerate = request.getParameterNames();
     final String matchString = groupName + "[";
 
     final Map<String, Map<String, String>> groupParam = new HashMap<>();
     while (enumerate.hasMoreElements()) {
       final String str = enumerate.nextElement();
       if (str.startsWith(matchString)) {
-        final int level1KeyEnd = str.indexOf("]");
+        try {
+          final int level1KeyEnd = str.indexOf("]");
```

Comment:
Can this be outside the try?

Reply:
If you mean the `level1KeyEnd` assignment no because then it will be out of scope inside the try block where it's used.

Reply:
Ah, missed that, thanks!